src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs

```diff
@@ -280,6 +283,24 @@ public bool HasPrivateKey
             }
         }
 
+        /// <summary>
+        /// Gets a bool that determines if the 'key_ops' (Key Operations) property should be serialized.
+        /// </summary>
+        /// <return>true if 'key_ops' (Key Operations) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeKeyOps()
+        {
+            return KeyOps.Any();
+        }
+
+        /// <summary>
+        /// Gets a bool that determines if the 'x5c' collection (X.509 Certificate Chain) property should be serialized.
+        ///</summary>
+        /// <return>true if 'x5c' collection (X.509 Certificate Chain) is not empty; otherwise, false.</return>
+        public bool ShouldSerializeX5c()
```

Comment:
Where is this used?

Reply:
See the comment above.

Reply:
oh, so does this puts the serialization under user control?